HorizonEngine/src/HorizonEngine/Camera.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#ifndef HZN_CAMERA_H
+#define HZN_CAMERA_H
+
+#include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/gtc/type_ptr.hpp>
+
+namespace Hzn
+{
+
+	class Camera
+	{
+	public:
+		Camera(
+			const glm::vec3& _position,
+			const glm::vec3& _front,
+			const glm::vec3& _up
+		): position(_position), front(_front), up(_up) {}
+
+		virtual glm::mat4 getViewMatrix() const = 0;
+		virtual glm::mat4 getProjectionMatrix() const = 0;
+		virtual ~Camera() = default;
+
+		glm::vec3 position = { 0.0f, 0.0f, 0.0f };
+		glm::vec3 front = { 0.0f, 0.0f, -1.0f };
+		glm::vec3 up = { 0.0f, 1.0f, 0.0f };
```

Comment:
What are these parameters

Reply:
These parameters, position up and front are the position of the camera In 3D space, and the two coordinates that of the camera (the front would be your z coordinate and up would be y coordinate). You can calculate the x coordinate by taking the cross product of these two coordinates.